File: python/grains/grains.py
import math


def square(number):
    if number < 1 or number > 64:
        raise ValueError("square must be between 1 and 64")
    return int(math.pow(2, number - 1))


def total():
    sum = 0
    for i in range(1, 65):
        sum += square(i)
    return sum

File: go/grains/grains.go
package grains

import "errors"

func Square(number int) (uint64, error) {
	if number < 1 || number > 64 {
		return 0, errors.New("invalid input")
	}
	if number == 1 {
		return 1, nil
	}
	halfRiceCount, err := Square(number - 1)
	if err != nil {
		return 0, errors.New("invalid input")
	}
	return halfRiceCount * 2, nil
}

func Total() uint64 {
	totalGrains := uint64(0)
	for i := 1; i <= 64; i++ {
		num, err := Square(i)
		if err != nil {
			return 0
		}
		totalGrains += num
	}
	return totalGrains
}
